internal/entities/payments/payments/core.go

```diff
@@ -1,12 +1,92 @@
 package payments
 
 import (
+	"errors"
 	"github.com/freshpay/internal/config"
+	"github.com/freshpay/internal/entities/payments/utilities"
+	"strings"
 )
 
-func AddPayments(payments *Payments) (err error) {
-	if err = config.DB.Table("payments").Create(payments).Error; err != nil {
+var InputPaymentsChannel =make(chan *Payments,1000)
+var ResultsPaymentsChannel=make(chan *Payments,1000)
+
+func AddPayments(payment *Payments) (err error) {
+	payment.Type=GetPaymentType(payment)
+	payment.Status="processing"
+	payment.ID="paymt_"+utilities.RandomString(14)
+	//err=ValidityCheck(payment)
+	//if err!=nil{
+	//	return err
+	//}
+	InputPaymentsChannel <-payment
+	if err= config.DB.Table("payments").Create(payment).Error; err != nil {
+		return err
+	}
+	return nil
+}
+
+func GetPaymentByID(payment *Payments, id string) (err error) {
+	if err = config.DB.Table("payments").Where("id = ?", id).First(payment).Error; err != nil {
```

Comment:
create a `payments/repo.go` and add DB query there. core shouldn't directly talk to DB